hooks/tk-multi-loader2/tk-3de4_actions.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that loads defines all the available actions, broken down by publish type.
+
+
+Originally from ``tk-multi-loader2/tk-maya_actions.py``, this file can be
+replaced by:
+
+1. Creating our own copy of ``tk-multi-loader2`` repository
+2. Edit the ``tk-multi-loader2/tk-maya_actions.py`` in there
+3. Point to our ``tk-multi-loader2`` repository in
+   ``env/includes/app_locations.yml:apps.tk-multi-loader2.location``
+"""
+import glob
+from itertools import groupby, izip
+import os
+import re
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+import tde4
+
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class FileExistenceError(Exception):
+    """
+    Exception when files don't exist on disk.
+    """
+    def __init__(self, path):
+        """
+        Initialise the class.
+
+        :param str path: The path that doesn't exist.
+        """
+        message = "Could not find file on disk for published file path: {!r}"
+        super(FileExistenceError, self).__init__(message.format(path))
```

Comment:
Subclass it from `OSError` so it is more specific, setting the `errno` to `ENOENT` to signify missing entity

```python
import errno
```


```suggestion
class FileExistenceError(OSError):
    """
    Exception when files don't exist on disk.
    """
    def __init__(self, path):
        """
        Initialise the class.

        :param str path: The path that doesn't exist.
        """
        message = "Couldn't find file on disk for published file path"
        super(FileExistenceError, self).__init__(errno.ENOENT, message, path)
```